csrgen.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env python3
+import argparse
+import fnmatch
+import re
+import os
+import subprocess
+import readline
+from jinja2 import Template, Environment, FileSystemLoader
+
+file_loader = FileSystemLoader('.')
+env = Environment(loader=file_loader)
+
+
+description = """
+Simple script to handle generating Certificate Signing Requests (CSR)
+This has the ability to generate CSRs with and without a provided key.
+Also can handle multiple domains as Subject Alternative Name (SAN) records.
+"""
+config_template = """[ req ]
+    default_bits           = 2048
+    distinguished_name     = req_distinguished_name
+    prompt                 = no
+    req_extensions         = v3_req
+[ req_distinguished_name ]
+    {% if c is not none -%}
+        C                      = {{s}}
+    {%- endif %}
+    {%- if l is not none -%}
+        L                      = {{l}}
+    {%- endif %}
+    {%- if s is not none -%}
+        ST                     = {{s}}
+    {%- endif %}
+    {%- if o is not none -%}
+        O                      = {{o}}
+    {%- endif %}
+{{'    CN                     = {{ cn }}
+[ v3_req ]
+    subjectAltName          = @alt_names
+
+[alt_names]
+    DNS.1 = {{ cn }}
+    {%- for domain in sans %}
+        {%- if domain %}
+    DNS.{{ loop.index +1 }} = {{ domain }}
+        {%- endif %}
+    {%- endfor %}
+'}}
+"""
+
+
+def check_file(file):
```

Comment:
The name `check_file` is ambiguous... what are you checking? Consider a name `file_exists` so your code reads more naturally, ex. `if file_exists():`

Reply:
Aaaand I just realized this is already a library function: `os.path.exists()` for any path, `pathlib.Path.is_file()` for files in particular.

Reply:
Also the `file` parameter is a bit ambiguous. Are you looking for a `file_name` with a string representation, a path-like object, an actual file object?

Can document in code by checking your parameter first thing in the code: `if not isinstance(file, str): raise TypeError`.

In general, good practice to check your parameters before using them, to make debugging easier.